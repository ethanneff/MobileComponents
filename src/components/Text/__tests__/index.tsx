import React from "react";
import DeviceInfo from "react-native-device-info";
import renderer, { act } from "react-test-renderer";
import Component from "..";

jest.useFakeTimers();
describe("Text Component", () => {
  it("renders on mobile", () => {
    const component = renderer.create(<Component title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders on tablet", () => {
    jest.spyOn(DeviceInfo, "isTablet").mockReturnValue(() => true);
    const component = renderer.create(<Component title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders centered", () => {
    const component = renderer.create(<Component center title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders centerVertically", () => {
    const component = renderer.create(
      <Component centerVertically title="test" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders h1", () => {
    const component = renderer.create(<Component h1 title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders h2", () => {
    const component = renderer.create(<Component h2 title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders h3", () => {
    const component = renderer.create(<Component h3 title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders h4", () => {
    const component = renderer.create(<Component h4 title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders h5", () => {
    const component = renderer.create(<Component h5 title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders body", () => {
    const component = renderer.create(<Component body title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders link", () => {
    const component = renderer.create(<Component link title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders error", () => {
    const component = renderer.create(<Component error title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders button", () => {
    const component = renderer.create(<Component button title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders clear", () => {
    const component = renderer.create(<Component clear title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders bold", () => {
    const component = renderer.create(<Component bold title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders uppercase", () => {
    const component = renderer.create(<Component uppercase title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders uppercase without title", () => {
    const component = renderer.create(<Component uppercase title="" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders ellipsis", () => {
    const component = renderer.create(<Component ellipsis title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders ellipsis numberOfLines", () => {
    const component = renderer.create(
      <Component ellipsis numberOfLines={2} title="test" />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders caption", () => {
    const component = renderer.create(<Component caption title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders tab", () => {
    const component = renderer.create(<Component tab title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders overline", () => {
    const component = renderer.create(<Component overline title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders as expected when hidden", () => {
    const component = renderer.create(<Component hidden title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders as expected when invisible", () => {
    const component = renderer.create(<Component invisible title="test" />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders notBold", () => {
    const component = renderer.create(<Component title="test" notBold />);
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders markdown", () => {
    const component = renderer.create(
      <Component title="normal *bold*" markdown />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders incremental", () => {
    const component = renderer.create(
      <Component title="44" incremental incrementBy={2} />
    );
    act(() => {
      jest.runAllTimers();
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
  it("renders incremental with props", () => {
    const component = renderer.create(
      <Component title="44" incremental incrementBy={2} />
    );
    act(() => {
      jest.runAllTimers();
    });
    expect(component.toJSON()).toMatchSnapshot();
  });
});
