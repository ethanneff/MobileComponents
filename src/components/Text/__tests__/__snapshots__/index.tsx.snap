// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text Component renders as expected when hidden 1`] = `null`;

exports[`Text Component renders as expected when invisible 1`] = `
<Text
  accessibilityLiveRegion="none"
  accessible={false}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "color": "transparent",
        },
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders body 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders bold 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        Object {
          "fontFamily": "BrandonText-Bold",
        },
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders button 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Bold",
          "fontSize": 15,
          "letterSpacing": 0.5,
          "lineHeight": 20,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders caption 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(199, 18%, 33%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 15,
          "letterSpacing": 0,
          "lineHeight": 20,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders centerVertically 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        Object {
          "flex": 1,
          "textAlignVertical": "center",
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders centered 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        Object {
          "textAlign": "center",
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders clear 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        Object {
          "color": "hsl(255, 100%, 100%)",
        },
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders ellipsis 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={1}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders ellipsis numberOfLines 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  ellipsizeMode="tail"
  numberOfLines={2}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders error 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(0, 65%, 51%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders h1 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Bold",
          "fontSize": 36,
          "letterSpacing": -0.1,
          "lineHeight": 48,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders h2 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Bold",
          "fontSize": 28,
          "letterSpacing": 0.3,
          "lineHeight": 36,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders h3 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Medium",
          "fontSize": 22,
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders h4 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Medium",
          "fontSize": 20,
          "letterSpacing": 0,
          "lineHeight": 28,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders h5 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Medium",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders incremental 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "hsl(200, 19%, 18%)",
        "fontFamily": "BrandonText-Regular",
        "fontSize": 17,
        "letterSpacing": 0,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  44
</Text>
`;

exports[`Text Component renders incremental with props 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "hsl(200, 19%, 18%)",
        "fontFamily": "BrandonText-Regular",
        "fontSize": 17,
        "letterSpacing": 0,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  44
</Text>
`;

exports[`Text Component renders link 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(176, 100%, 25%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders markdown 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "hsl(200, 19%, 18%)",
        "fontFamily": "BrandonText-Regular",
        "fontSize": 17,
        "letterSpacing": 0,
        "lineHeight": 24,
      },
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    style={
      Array [
        Array [
          Object {
            "color": "hsl(200, 19%, 18%)",
            "fontFamily": "BrandonText-Regular",
            "fontSize": 17,
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        undefined,
      ]
    }
    textBreakStrategy="simple"
  >
    normal 
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    style={
      Array [
        Array [
          Object {
            "color": "hsl(200, 19%, 18%)",
            "fontFamily": "BrandonText-Regular",
            "fontSize": 17,
            "letterSpacing": 0,
            "lineHeight": 24,
          },
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
          undefined,
        ],
        Object {
          "fontFamily": "BrandonText-Bold",
        },
      ]
    }
    textBreakStrategy="simple"
  >
    bold 
  </Text>
</Text>
`;

exports[`Text Component renders notBold 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        Object {
          "fontFamily": "BrandonText-Regular",
        },
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders on mobile 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders on tablet 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders overline 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(199, 18%, 33%)",
          "fontFamily": "BrandonText-Medium",
          "fontSize": 14,
          "letterSpacing": 1.5,
          "lineHeight": 20,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  TEST
</Text>
`;

exports[`Text Component renders tab 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(199, 18%, 33%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 14,
          "letterSpacing": 0,
          "lineHeight": 16,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  test
</Text>
`;

exports[`Text Component renders uppercase 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  style={
    Array [
      Array [
        Object {
          "color": "hsl(200, 19%, 18%)",
          "fontFamily": "BrandonText-Regular",
          "fontSize": 17,
          "letterSpacing": 0,
          "lineHeight": 24,
        },
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
        undefined,
      ],
      undefined,
    ]
  }
  textBreakStrategy="simple"
>
  TEST
</Text>
`;

exports[`Text Component renders uppercase without title 1`] = `null`;
